enterprise/api/v1/serializers.py

```diff
@@ -330,12 +331,19 @@ def save(self):  # pylint: disable=arguments-differ
         Save the EnterpriseCustomerUser.
         """
         enterprise_customer = self.validated_data['enterprise_customer']
+        active = self.validated_data.get('active', True)
+        defaults = {'active': active}
 
-        ecu = models.EnterpriseCustomerUser(
+        __, created = models.EnterpriseCustomerUser.objects.update_or_create(
             user_id=self.user.pk,
             enterprise_customer=enterprise_customer,
+            defaults=defaults,
         )
-        ecu.save()
+        if not created and active:
+            # if learner has activated enterprise we need to de-activate other enterprises learner is linked to
+            models.EnterpriseCustomerUser.objects.filter(
+                user_id=self.user.pk
+            ).exclude(enterprise_customer=enterprise_customer).update(active=False)
```

Comment:
This needs to cater for a user **concurrently** logging in from multiple enterprises using different browser sessions.